src/core/ext/transport/chttp2/transport/context_list.h

```diff
@@ -32,18 +31,46 @@ namespace grpc_core {
 class ContextList {
  public:
   // Creates a new element with \a context as the value and appends it to the
-  // list.
-  static void Append(ContextList** head, grpc_chttp2_stream* s);
+  // list. Each RPC/stream is associated with a unique \a context. This method
+  // is invoked when a chunk of data stored in an outgoing buffer is going to be
+  // sent over the wire. A data chunk being written over the wire is multiplexed
+  // with bytes from multiple RPCs. If one such RPC is traced, we store the
+  // following information about the traced RPC:
+  // - byte_offset_in_stream: Number of bytes belonging to that traced RPC which
+  // have been sent so far from the start of the RPC stream.
+  // - relative_start_pos_in_chunk: Starting offset of the traced RPC within
+  // the current chunk that is being sent.
+  // - num_traced_bytes_in_chunk: Number of bytes belonging to the traced RPC
+  // within the current chunk.
+  static void Append(ContextList** tail, void* context,
```

Comment:
This API is very confusing & C-like... if we're making changes can we consider doing something a little simpler here?

I'd recommend gathering up all the elements in a `std::vector` so that we get some better cache coherency properties than what's here... once we do that we can be talking about just a `std::vector<ContextEntry>` with `ContextEntry` containing our handful of members and no complicated pointer chasing code.

Reply:
updated it to use std::vector type. It also required some changes to chttp2_transport